java/ql/src/experimental/Security/CWE/CWE-094/TemplateInjection.qll

```diff
@@ -0,0 +1,247 @@
+/** Definitions related to the Server Side Template Injection (SSTI) query. */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.FlowSources
+import experimental.semmle.code.java.frameworks.FreeMarker
+import experimental.semmle.code.java.frameworks.Velocity
+import experimental.semmle.code.java.frameworks.MVEL
+import experimental.semmle.code.java.frameworks.JinJava
+import experimental.semmle.code.java.frameworks.Pebble
+import experimental.semmle.code.java.frameworks.Thymeleaf
+
+/** A module to reason about Server Side Template Injection (SSTI) vulnerabilities */
+module TemplateInjection {
+  /** A taint tracking configuration to reason about Server Side Template Injection (SSTI) vulnerabilities */
+  class TemplateInjectionFlowConfig extends TaintTracking::Configuration {
+    // import TemplateInjectionCustomizations::TemplateInjection;
+    TemplateInjectionFlowConfig() { this = "TemplateInjectionFlowConfig" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    override predicate isSanitizer(DataFlow::Node node) {
+      node.getType() instanceof PrimitiveType or node.getType() instanceof BoxedType
+    }
+
+    override predicate isAdditionalTaintStep(DataFlow::Node prev, DataFlow::Node succ) {
+      exists(AdditionalFlowStep a | a.isAdditionalTaintStep(prev, succ))
+    }
+  }
+
+  /**
+   * A data flow source for `Configuration`.
+   */
+  private class Source extends DataFlow::Node {
+    Source() { this instanceof RemoteFlowSource or this instanceof LocalUserInput }
```

Comment:
Does local input make sense for this query?

Reply:
removed.